packages/design-system-tokens/src/lib/exportJson.ts

```diff
@@ -0,0 +1,42 @@
+import { FileDescriptor } from './types';
+import { writeFile } from './utility';
+
+/*
+ * Accepts an array of file descriptors generated by getFileDescriptors('path')
+ * and writes their imported data to filesystem as a JSON file
+ */
+export const exportJson = (fileDescriptors: FileDescriptor[], outPath: string): number => {
+  let filename = '';
+  let isTheme = false;
+  const tokenOutput: string | any = {};
+
+  fileDescriptors.forEach((file) => {
+    /*
+     * doing this asynchronously with import() makes this code needlessly complex
+     * so ignoring this particular linting error here to allow named require.
+     * which runs synchronously.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const importedModule = require(`${file.moduleImportName}`);
+
+    tokenOutput[file.fileBaseName] = {
+      ...importedModule.default,
+    };
+
+    // theme files have a description prop token files do not
+    if (importedModule.default.description !== undefined) {
+      isTheme = true;
+      tokenOutput.tokenType = 'theme';
+      filename = `${outPath}/${file.parentDirectoryName}-${file.fileBaseName}.tokens.json`;
+      writeFile(filename, JSON.stringify(tokenOutput, null, 4));
+    } else {
+      // it's a token file
+      tokenOutput.tokenType = 'tokens';
+      filename = `${outPath}/cmsds.tokens.json`;
+    }
+  });
+  if (!isTheme) writeFile(filename, JSON.stringify(tokenOutput, null, 4));
```

Comment:
[question] I'm confused by the `isTheme` variable. Will the `fileDescriptors` list ever have a list of files that are of mixed type (some themes & some note themes)?

If so, does the `else` conditional in the `forEach` need to set the `isTheme` variable back to false at some point in order to trigger this line?

Reply:
No it only runs on one type of 'tokens' or 'themes' at a time so this wont need to be reset. Themes vary slightly from tokens in their structure and require different processing which is why this is necessary, similar code in all the exporters.